### ISSUE TITLE
Release Sim2Reason artifacts (models, datasets) on Hugging Face

### ISSUE THREAD

**Issue by NielsRogge** [Sim2Reason/Sim2Reason#1]
Hi @Aryan-Satpathy 🤗

Niels here from the open-source team at Hugging Face. I discovered your work through Hugging Face's daily papers as yours got featured: https://huggingface.co/papers/2604.11805.
The paper page lets people discuss about your paper and lets them find artifacts about it (your models, datasets or demo for instance), you can also claim
the paper as yours which will show up on your public profile at HF, add Github and project page URLs.

I noticed in your GitHub README (https://github.com/Sim2Reason/Sim2Reason) that you're planning to release your pretrained checkpoints for various Qwen models, as well as the IPhO validation set, on the 🤗 Hub, but the links are currently placeholders. I also saw that your method generates new synthetic QA datasets (e.g., `train_v1_rl.parquet`, `test_v1_rl.parquet`). That's fantastic news!

It'd be great to make these checkpoints and datasets officially available on the 🤗 Hub, to improve their discoverability and visibility. We can add tags so that people find them when filtering https://huggingface.co/models and https://huggingface.co/datasets.

## Uploading models

See here for a guide: https://huggingface.co/docs/hub/models-uploading.

In this case, we could leverage the [PyTorchModelHubMixin](https://huggingface.co/docs/huggingface_hub/package_reference/mixins#huggingface_hub.PyTorchModelHubMixin) class which adds `from_pretrained` and `push_to_hub` to any custom `nn.Module`. Alternatively, one can leverages the [hf_hub_download](https://huggingface.co/docs/huggingface_hub/en/guides/download#download-a-single-file) one-liner to download a checkpoint from the hub.

We encourage researchers to push each model checkpoint to a separate model repository, so that things like download stats also work. We can then also link the checkpoints to the paper page.

## Uploading dataset

Your generated synthetic QA datasets and the IPhO validation set seem like valuable resources! Would be awesome to make them available on 🤗 , so that people can do:

```python
from datasets import load_dataset

dataset = load_dataset("your-hf-org-or-username/your-dataset")
```
See here for a guide: https://huggingface.co/docs/datasets/loading.
For your synthetic data, since it's in parquet format, it would be very easy to upload!

Besides that, there's the [dataset viewer](https://huggingface.co/docs/hub/en/datasets-viewer) which allows people to quickly explore the first few rows of the data in the browser.

Let me know if you're interested/need any help regarding this!

Cheers,

Niels
ML Engineer @ HF 🤗

**Comment by NielsRogge** [comment]
Hey @Aryan-Satpathy, it's great to see that you've uploaded the artifacts to the Hugging Face hub.

However, it seems they are not yet linked to the paper page. Could you link them by mentioning the paper URL (preferably https://huggingface.co/papers/2604.11805, otherwise the arXiv abstract or PDF URL) in the README of the Hugging Face model/dataset/Space repos?
This will make the artifacts automatically show up on the paper page.

See also here for more information: https://huggingface.co/docs/hub/model-release-checklist#write-a-comprehensive-model-card (in case of models) or https://huggingface.co/docs/hub/en/datasets-cards (in case of datasets)

Thanks

**Comment by Aryan-Satpathy** [comment]
Thanks for the heads up! All the models, datasets, and the collection have been linked to the paper page. I have also submitted authorship claim for [huggingface.co/papers/2604.11805](https://huggingface.co/papers/2604.11805), which is pending.

**Comment by NielsRogge** [comment]
Thanks so much for uploading all the artifacts to the Hugging Face Hub and linking them to the paper page. The collection looks great, and it's wonderful to see everything properly organized and accessible.

I also see you've submitted an authorship claim for the paper page - once that's approved, the paper will show up on your Hugging Face profile as well.

Thanks again for making your work openly available to the community!
